### ISSUE TITLE
[Follow up to #36034] Website logo is not restored when removing custom filter during `Network blocking` testing

### ISSUE THREAD

**Issue by hffvld** [brave/brave-browser#39680]
### Description

Follow-up to #36034.

### Steps to reproduce

1. Launch Brave
2. Go to https://www.iana.org/help/example-domains > *Confirm that `logo` is shown*
3. Go to Brave's `Settings` > Shileds & Privacy
4. Content Filtering > CUSTOM FILTERS
5. Insert text from below and `Save`:
```
! Hide the header on example.com (CF Test)
example.com,example.net##h1

! Hide the logo on iana.org/help/example-domains (Network test)
||iana.org/_img/2022/iana-logo-header.svg

! Hide the brave logo on brave.com (Network test)
||brave.com/static-assets/images/brave-logo-sans-text.svg
```
6. Close the settings and refresh https://www.iana.org/help/example-domains > *Confirm that `logo` is not shown*
7. Now remove the custom filter and check https://www.iana.org/help/example-domains once again > *Observe that `logo` is still not shown*

### Actual result

The website's logo is not restored when removing the custom filter during `Network blocking` testing.

-----------------

https://github.com/brave/brave-browser/assets/128532965/450b82cd-e4cb-4f38-af84-47eda4e703e9

### Expected result

The website's logo must be shown again after a custom filter is removed during `Network blocking` testing.

----------------

https://github.com/brave/brave-browser/assets/128532965/ec3b85c1-69d7-40a2-8880-432702ad2157

### Reproduces how often

Easily reproduced

### Brave version

Brave build: 1.68 (115)
BraveCore: 1.68.115 (127.0.6533.26)

### Device/iOS version

iPhone 14 / iOS 17.6

### Affected browser versions

- [ ] latest AppStore
- [X] latest TestFlight
- [ ] previous TestFlight

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] in the latest version of mobile Safari

### Miscellaneous information

cc @cuba

**Comment by cuba** [comment]
The issue here is that when you clear the whole list the content blocker is not removed. If you modify the list then it updates correctly. The fix is fairly straightforward.

**Comment by kjozwiak** [comment]
Closing as the above was fixed via https://github.com/brave/brave-core/pull/24713 which was merged directly into `1.68.x`. The above requires `1.68.125` or higher for `1.68.x` verification 👍

**Comment by hffvld** [comment]
Verified on `iPhone 14` using version(s):
```
Device/OS: iPhone 14 / iOS 17.6
Brave build: 1.68 (125)
BraveCore: 1.68.125 (127.0.6533.57)
```
---------------------------

**STEPS:**
1. Follow the STR/TP from https://github.com/brave/brave-browser/issues/39680#issue-2401880014
2. *Verify*

**ACTUAL RESULTS:**
- Verified that the website's logo is shown again after removing a custom filter.

---------------------------

https://github.com/user-attachments/assets/a73be263-4b49-4cb2-a2dc-43fd823bc507
